src/geometry.h

```diff
@@ -29,8 +29,32 @@ struct Point {
 		this->y = y;
 	}
 
+	/**
+	 * Copy constructor.
+	 * @param p %Point to copy.
+	 */
+	template<typename CT2>
+	Point(const Point<CT2> &p)
+	{
+		this->x = p.x;
+		this->y = p.y;
+	}
+
 	typedef CT CoordType; ///< Type of the coordinate value.
 
+	/**
+	 * Add a 2D point to this one.
+	 * @param q Second point to add.
+	 * @return Both dimensions added to each other.
+	 */
+	template<typename CT2>
+	inline Point<CT> operator+=(const Point<CT2> &q)
```

Comment:
Wouldn't it make sense to return a reference rather than a copy of the value?

I tried to find the rule about this, but failed. I did find examples, but not a discussion.